ts-client/src/amm/index.ts

```diff
@@ -827,15 +835,17 @@ export default class AmmImpl implements AmmImplementation {
     const { provider, vaultProgram, ammProgram } = createProgram(connection, opt?.programId);
 
     const poolState = await getPoolState(pool, ammProgram);
-
-    invariant(tokenInfoA.address === poolState.tokenAMint.toBase58(), `TokenInfoA provided is incorrect`);
-    invariant(tokenInfoB.address === poolState.tokenBMint.toBase58(), `TokenInfoB provided is incorrect`);
-    invariant(tokenInfoA, `TokenInfo ${poolState.tokenAMint.toBase58()} A not found`);
-    invariant(tokenInfoB, `TokenInfo ${poolState.tokenBMint.toBase58()} A not found`);
+    const { tokenAMint, tokenBMint } = poolState
+    const [tokenASupply, tokenBSupply] = await Promise.all([
+      provider.connection.getTokenSupply(tokenAMint),
+      provider.connection.getTokenSupply(tokenBMint)
+    ])
+    const tokenADecimals = tokenASupply.value.decimals
+    const tokenBDecimals = tokenBSupply.value.decimals
```

Comment:
can we do things similar like `vault-sdk` ? use `getMint` instead and pass the whole `Mint` to AmmImpl instance